subschemas/substrate-api-proxy/src/resolvers/Query/treasury.ts

```diff
@@ -28,19 +30,41 @@ export async function treasurySummary(
 
   return {
     activeProposals: proposals.proposals.length,
-    proposalCount: proposals.proposalCount.toString(),
+    totalProposals: proposals.proposalCount.toNumber(),
     approvedProposals: proposals.approvals.length,
-    spendPeriod: api.consts.treasury.spendPeriod.toString(),
+    spendPeriod: createSpendPeriod(api, bestNumber),
     treasuryBalance: {
       accountId: treasuryBalance.accountId.toString(),
       accountNonce: treasuryBalance.accountNonce.toString(),
-      freeBalance: treasuryBalance.freeBalance.toString(),
-      frozenFee: treasuryBalance.frozenFee.toString(),
-      frozenMisc: treasuryBalance.frozenMisc.toString(),
-      reservedBalance: treasuryBalance.reservedBalance.toString(),
-      votingBalance: treasuryBalance.votingBalance.toString(),
+      freeBalance: formatBalance(api, treasuryBalance.freeBalance),
+      frozenFee: formatBalance(api, treasuryBalance.frozenFee),
+      frozenMisc: formatBalance(api, treasuryBalance.frozenMisc),
+      reservedBalance: formatBalance(api, treasuryBalance.reservedBalance),
+      votingBalance: formatBalance(api, treasuryBalance.votingBalance),
     },
-    burn: burn?.toString(),
+    nextBurn: formatBalance(api, burn ?? BN_ZERO, true),
+  };
+}
+
+function createSpendPeriod(api: Context['api'], bestNumber: BlockNumber): SpendPeriod {
```

Comment:
Should this go in the `substrateChainService`?

Reply:
I don't think we will use this format in other places, the value of the spendPeriod itself may be used in the bounties but not in this shape. We can move it later if needed